src/overview/tabs/Details/cards/VmMigrationsHistoryChart.tsx

```diff
@@ -0,0 +1,166 @@
+      setChartDimensions({ height, width });
+    }
+  };
+
+  useLayoutEffect(() => {
+    if (chartContainerRef.current) {
+      const resizeObserver = getResizeObserver(chartContainerRef.current, handleResize, true);
+      handleResize();
+
+      return () => {
+        resizeObserver();
+      };
+    }
+    return undefined;
+  }, []);
+
+  const maxVmMigrationValue = Math.max(
+    ...vmMigrationsDataPoints.running.map((migration) => migration.value),
+    ...vmMigrationsDataPoints.failed.map((migration) => migration.value),
+    ...vmMigrationsDataPoints.succeeded.map((migration) => migration.value),
+  );
+
+  const legendData = [
+    { name: t('Running'), symbol: { fill: ChartColors.Running } },
+    { name: t('Failed'), symbol: { fill: ChartColors.Failure } },
+    { name: t('Succeeded'), symbol: { fill: ChartColors.Success } },
+  ];
+
+  const maxTicks = Math.max(5, Math.ceil(maxVmMigrationValue) + 1);
+  const tickValues = Array.from({ length: maxTicks }, (_, i) => i + 1);
+
+  return (
+    <div ref={chartContainerRef} className="pf-u-h-100 pf-u-w-100">
+      <Chart
+        ariaDesc={t('Area chart with VM migration statistics')}
+        containerComponent={
+          <ChartVoronoiContainer
+            labels={({ datum }: { datum: ChartDatumWithName }) =>
+              datum.y === 0 || datum.name.includes('scatter')
+                ? (undefined as unknown as string)
+                : `${datum.x} - ${datum.name ?? 'Unknown'}: ${datum.y ?? 0}`
+            }
+            constrainToVisibleArea
+          />
+        }
+        legendData={legendData}
+        legendPosition="bottom"
+        maxDomain={{ y: maxVmMigrationValue ? undefined : 5 }}
+        padding={{
+          bottom: 55,
+          left: 50,
+          right: 50,
+          top: 20,
+        }}
+        width={chartDimensions.width}
+        height={chartDimensions.height}
+      >
+        <ChartAxis />
+        <ChartAxis dependentAxis showGrid tickValues={tickValues} />
+        <ChartGroup>
+          <ChartArea
+            data={
+              vmMigrationsDataPoints.succeeded.map(({ dateLabel, value }) => ({
+                name: t('Succeeded'),
+                x: dateLabel,
+                y: value,
+              })) as ChartDatumWithName[]
+            }
+            colorScale={[ChartColors.Success]}
+          />
+          <ChartScatter
+            data={
+              vmMigrationsDataPoints.succeeded.map(({ dateLabel, value }) => ({
+                name: 'Succeeded scatter',
+                x: dateLabel,
+                y: value,
+              })) as ChartDatumWithName[]
+            }
+            size={3}
+            symbol="circle"
+            colorScale={[ChartColors.Success]}
+          />
+          <ChartArea
+            data={
+              vmMigrationsDataPoints.failed.map(({ dateLabel, value }) => ({
+                name: t('Failed'),
+                x: dateLabel,
+                y: value,
```

Comment:
This map function is the same about 6 times, let's extract it to get the array and the name